cpmpy/transformations/linearize.py

```diff
@@ -572,3 +622,39 @@ def only_positive_coefficients(lst_of_expr):
 
     return newlist
 
+def linearize_objective(expr, supported=frozenset(["sum","wsum"])):
+    """
+    Takes a linear expression and outputs:
+    
+    - Decision variable: Var
+    - Linear: sum([Var])                                   (CPMpy class 'Operator', name 'sum')
+              wsum([Const],[Var])                          (CPMpy class 'Operator', name 'wsum')
+
+    Only difference with flatten_objective is that no negated boolvars are allowed.
+    """
+    
+    flatexpr, flatcons = flatten_objective(expr, supported=supported)
+    
+    if isinstance(flatexpr, _NumVarImpl) or flatexpr.name in {"sum","wsum"}:
+        pos_expr, const = only_positive_bv_sub(flatexpr)
+        if (const != 0):
+            assert isinstance(pos_expr, Operator) and pos_expr.name == "wsum", f"unexpected expression, should be wsum but got {pos_expr}"
+            pos_expr = Operator("wsum", [pos_expr.args[0]+[1], pos_expr.args[1] + [const]])
+        return pos_expr, flatcons
+    
+    else: 
+        if flatexpr.name not in supported:
+            raise Exception(f"{flatexpr} is not linear or is not supported. Please report on github")
```

Comment:
minor: I would prefer to refactor this to:

```python
...
else:
    assert flatexpr.name in supported, ("Unexpected numerical expression, {flatexpr} is not linear and not supported, please report on github)
   ...
```

As this means something went wrong in flatten_objective